core/src/validator.rs

```diff
@@ -189,24 +189,19 @@ impl Validator {
                     std::process::exit(1);
                 }
                 let bank = bank_forks.read().unwrap().working_bank();
-                let stake: Option<u64>> = bank
-                    .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0)
-                    .iter()
-                    .find(|account| match (account) {
-                        Some(pubkey, (stake, account)) => {
-                            if *pubkey == keypair.pubkey() && *stake > 0 {
-                                if let Ok(vote_state) = VoteState::deserialize(&account.data) {
-                                    return Some(*stake);
-                                } 
-                            }
-                            return None;
-                        },
+                let stake: Option<u64> = bank
+                    .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0) //Option<&HashMap<Pubkey, (u64, Account)>>
+                    .map(|account| {
+                        account.iter().find(|(pubkey, (stake, account))| {
+                            *pubkey == &keypair.pubkey()
+                                && *stake > 0
+                                && Ok(vote_state) == VoteState::deserialize(&account.data)
```

Comment:
@mvines   I got this error here ` Ok(vote_state) == VoteState::deserialize(&account.data) ) 
    |                                ^^^^^^^^^^ not found in this scope`   How can I fix it?

Reply:
You'll need to bring it into scope with `use solana_vote_api::vote_state::VoteState;`

Reply:
@mvines   it is already [here](https://github.com/cyberbono3/solana/blob/warn-validators/core/src/validator.rs#L33)

Reply:
Is it maybe the `vote_state` variable that's underlined in the error message?
I think this line can be instead:
```
&& VoteState::deserialize(&account.data).is_ok()
```

Reply:
@CriesofCarrots  thank you